starport_template/lib/starport_app.dart

```diff
@@ -22,16 +24,18 @@ class StarportApp extends StatelessWidget {
   static late NetworkInfo networkInfo;
   static late SecureDataStore secureDataStore;
   static late SettingsStore settingsStore;
+  static late ThemeStore themeStore;
   static late CosmosAuth cosmosAuth;
 
   @override
   Widget build(BuildContext context) {
-    return CosmosTheme(
-      child: MaterialApp(
-        title: 'Starport template',
-        //darkTheme: CosmosTheme.buildDarkAppTheme(), // enable for dark mode
-        theme: CosmosTheme.buildAppTheme(),
-        home: const RoutingPage(),
+    return Observer(
+      builder: (context) => CosmosTheme(
+        themeData: themeStore.isDarkTheme ? CosmosTheme.darkThemeData : const CosmosThemeData(),
```

Comment:
could we implement the suggestion from this comment first:
https://github.com/tendermint/flutter/pull/266#issuecomment-1118337803

Also, we could get rid of this discrepancy of having light theme in default constructor and dark theme in static field. instead, lets have separate constructors for those:
```dart
CosmosThemeData.light()
CosmosThemeData.dark()
```

then, we could specify this piece as:

```dart
Observer(
    builder: (context) => CosmosTheme(
        brightness: () => themeStore.isDarkTheme ? Brightness.dark : Brightness.light,
        // by default, CosmosTheme would have default params set up for light and dark theme, so we don't need to provide anything here
    )
)
```

and the `CosmosTheme` constructor:
```dart
const CosmosTheme({
    this.themeData = CosmosThemeData.light(),
    this.darkThemeData = CosmosThemeData.dark(),
    ...
})